python/rola/core.py

```diff
@@ -0,0 +1,52 @@
+from radix_engine_toolkit import derive_virtual_account_address_from_public_key, PublicKey
+
+from rola.models.signed_challenge import SignedChallenge
+from rola.utils.gateway import get_entity_owner
+from rola.utils.helpers import create_public_key_hash
+from rola.utils.ret import derive_address
+
+
+class Rola:
+
+    def __init__(self,
+                 expected_origin: str,
+                 dapp_address: str,
+                 application_name: str,
+                 network_id: str
+                 ):
+        self.expected_origin = expected_origin
+        self.dapp_address = dapp_address
+        self.application_name = application_name
+        self.network_id = network_id
+
+    def verify_signed_challenge(self,
+                                signed_challenge: SignedChallenge) -> bool:
+
+        # create public key hex hash
+        public_key_hex = create_public_key_hash(
```

Comment:
Minor: should this variable be called `public_key_hash_hex` instead of `public_key_hex`?

Reply:
yes